dbms/src/DataTypes/DataTypesDecimal.h

```diff
@@ -332,12 +332,12 @@ convertToDecimal(const typename FromDataType::FieldType & value, UInt32 scale)
         {
             static constexpr __int128 min_int128 = __int128(0x8000000000000000ll) << 64;
             static constexpr __int128 max_int128 = (__int128(0x7fffffffffffffffll) << 64) + 0xffffffffffffffffll;
-            if (out < min_int128 || out > max_int128)
+            if (! (out > min_int128 && out < max_int128))
```

Comment:
Let's make it in more general way. There're greaterOp() functions in AccurateComparison.h. If we do not care about performance too much, it's better to use them. At least we avoid redefinition of constants here.